### PR TITLE
TRAX-25: First version of glossary adding and with little refactoring.

### PR DESCRIPTION
**Pull request by edwardhura**
[TRAX-25](https://verbit.atlassian.net/browse/TRAX-25)

Added possibility to add new terms to glossary and with Glossary Term refactors over the project.

### CODE REVIEW

File: src/components/Editor/Editor.tsx [hunk 1]
```diff
@@ -149,7 +149,7 @@ function createBlocks(segment: Segment, { editable }: { editable: boolean }): Bl
         return chunks
     }, [])
 
-    return paragraphs.map(p => {
+    return paragraphs.map((p) => {
```
Comment (bartzy) [inline src/components/Editor/Editor.tsx:152]: why is it happening? Is this the prettier plugin?

File: src/network/HttpClient.ts [hunk 2]
```diff
@@ -67,7 +68,11 @@ export class HttpClient implements TransportLayer {
 
     async getGlossary(): Promise<Glossary> {
         const response = await this.client.get<GlossaryJSON>('/gloss')
-        const glossary = toGlossary(response.data)
-        return glossary
+        return toGlossary(response.data)
+    }
+
+    async addGlossary(data: TermRequestBody): Promise<GlossaryTerm> {
```
Comment (bartzy) [inline src/network/HttpClient.ts:74]: let's rename this to `addGlossaryTerm`

File: src/components/Editor/Editor.tsx (hunk 1 above)
Comment (edwardhura) [reply src/components/Editor/Editor.tsx:152]: yes

Comment (bartzy) [reply src/components/Editor/Editor.tsx:152]: Yeah they changed the default behavior a couple of weeks ago. Oh well :)

**Pull request merged**